File: services/authentication-server/src/main/java/com/balamut/authenticationserver/user/UserService.java
package com.balamut.authenticationserver.user;

import com.balamut.authenticationserver.jwt.TokenType;
import com.balamut.authenticationserver.user.exception.UserException;
import com.balamut.authenticationserver.user.request.RegisterRequest;
import com.balamut.authenticationserver.user.request.UserRequest;
import com.balamut.authenticationserver.user.response.EmailResponse;
import com.balamut.authenticationserver.user.response.RegisterResponse;
import com.balamut.authenticationserver.user.response.UserResponse;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface UserService {

    RegisterResponse register(RegisterRequest request) throws UserException;

    boolean matchesCredentials(String email, String password);

    String generateToken(String email, TokenType type) throws UserException;

    UserResponse getCurrentUser() throws UserException;

    List<UserResponse> getUsers(String role);

    void deleteUser(Integer id) throws UserException;

    UserResponse getUser(Integer id) throws UserException;

    EmailResponse getEmailInformation(String email);

    void changePassword(String oldPassword, String password);

    ResponseEntity<Void> changeUser(Integer id, UserRequest request);
}

File: application/src/app/services/user/user.service.ts
import {User} from "./user";
import {UserEmail} from "./user-email";

export interface UserService {
  getUserById: (id: number) => Promise<User>;
  changeUserById: (id: number, user: User) => Promise<boolean>;
  deleteUserById: (id: number) => Promise<boolean>;
  changePassword: (password: string, oldPassword: string) => Promise<boolean>;
  register: (info: {firstname: string, lastname: string, email: string, password: string}) => Promise<boolean>;
  getAllUsers: (role: string) => Promise<User[]>;
  getCurrentUser: () => Promise<User>;
  getInformationByEmail: (email: string) => Promise<UserEmail>;
}
